achievements/serializers.py

```diff
@@ -7,10 +7,10 @@ class AchievementSerializer(serializers.ModelSerializer):
     class Meta:
         model = Achievement
         fields = "__all__"
-        depth = 0
```

Comment:
Question: what does the depth do?

Reply:
It's used for foreign keys/many-to-many relations. When the depth is 1, instead of obtaining the foreign key when requesting the object, the associated object is obtained instead.
If the depth is 2, then associated objects of associated objects are loaded in, etc. (Not sure how this interacts with many-to-many relations though)

Reply:
It makes sure that it is possible to see the name and description of the associated category. With a depth of 0 you can only get the id of that category.

Reply:
That is an interesting way to solve the N+1 query problem: instead of lazy-loading items, always load them or never load them depending on the model configuration.